src/httpinterface.h

```diff
@@ -10,14 +11,14 @@
 #include "utilities/utils.h"
 
 struct HttpResponse {
-  HttpResponse(const std::string &body_in, const long http_status_code_in, CURLcode curl_code_in,
-               const std::string &error_message_in)
-      : body(body_in),
+  HttpResponse(std::string body_in, const int64_t http_status_code_in, CURLcode curl_code_in,
+               std::string error_message_in)
+      : body(std::move(body_in)),
         http_status_code(http_status_code_in),
         curl_code(curl_code_in),
-        error_message(error_message_in) {}
+        error_message(std::move(error_message_in)) {}
   std::string body;
-  long http_status_code;
+  int64_t http_status_code;
```

Comment:
I recently changed this to `long` because that is what the curl interface returns and that's the only place we get it from. Why would we want to use anything other than that?

Reply:
because long is not size defined. So it is not crossplatform. But int64_t is always 64 bit integer

Reply:
Right, but if the curl headers use long, the safest option for us is to use the same thing, since on one machine, they will compile to the same thing. If long != int64_t on some machine, then we may have compatibility problems. Unlikely to cause an actual problem, but possible.